src/main.rs

```diff
@@ -62,131 +60,18 @@ main!(|args: Cli| {
         project_dir.display()
     );
 
-    let _template = RepoBuilder::new()
-        .bare(false)
-        .with_checkout(CheckoutBuilder::new())
-        .clone(&args.git, &project_dir)
-        .with_context(|_e| format!("Couldn't clone `{}`", &args.git))?;
-
-    remove_dir_all(&project_dir.join(".git")).context("Error cleaning up cloned template")?;
-
-    let engine = liquid::ParserBuilder::new().build();
-    let mut placeholders = liquid::Object::new();
-    placeholders.insert(String::from("project-name"), liquid::Value::scalar(&name));
-    placeholders.insert(
-        String::from("crate_name"),
-        liquid::Value::scalar(&ident_case::RenameRule::SnakeCase.apply_to_field(&name)),
-    );
-    placeholders.insert(
-        String::from("authors"),
-        liquid::Value::scalar(&get_authors()?),
-    );
-
-    let progress = indicatif::ProgressBar::new_spinner();
-    progress.tick();
+    git::create(&project_dir, args)?;
+    git::remove_history(&project_dir)?;
 
-    for entry in WalkDir::new(&project_dir) {
-        let entry = entry?;
-        if entry.metadata()?.is_dir() {
-            continue;
-        }
+    let mut template = template::new();
+    template = template::substitute(&name, template)?;
```

Comment:
For what is currently happening here the `template::new()` is confusing -- it is a (constant) constructor of an empty object that is immediately passed to `substitute`. And `substitute` is only ever called with this (still constant) value.

I'd change this whole thing to `let template = template::substitute(&name);`

Reply:
wfm